core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -97,14 +106,15 @@ public static boolean isIngestNode(Settings settings) {
      * and updated.
      * </p>
      *
-     * @param nodeId     the nodes unique id.
-     * @param address    the nodes transport address
-     * @param attributes node attributes
-     * @param roles      node roles
-     * @param version    the version of the node.
+     * @param id               the nodes unique (ephemeral and persistent) node id
```

Comment:
In general I find these overloaded constructors dangerous, making it highly likely that some places are using the wrong one or unaware of the implications of generating a fresh ephemeral id.